### PR TITLE
Migrate ballerina-sql module to separate repository

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
$subject as part of task https://github.com/ballerina-platform/ballerina-lang/issues/25085

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
HSQL

### CODE REVIEW

File: sql-ballerina/build.gradle [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+task ballerinaBuild {
+    dependsOn(unpackJballerinaTools)
+    dependsOn(":sql-native:build")
+    dependsOn(test)
+    dependsOn(updateTomlFile)
+    finalizedBy(revertTomlFile)
+
+    inputs.dir file(project.projectDir)
+    doLast {
```
Comment (MadhukaHarith92) [inline sql-ballerina/build.gradle:135]: Shall we enable this config as well: https://github.com/ballerina-platform/module-ballerina-xmlutils/blob/master/xmlutils-ballerina/build.gradle#L74

**Review by MadhukaHarith92 — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        html.destination = new File("${buildDir}/reports/jacoco/report.html")
+        csv.destination = new File("${buildDir}/reports/jacoco/report.csv")
+    }
+
+    onlyIf = {
+        true
+    }
+}
+
+description = 'Ballerina - SQL'
```
Comment (daneshk) [inline build.gradle:100]:
```suggestion
description = 'Ballerina - SQL'

```

File: sql-ballerina/src/sql/tests/pool/connection-pool-test.bal [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+    test:assertTrue(dbError is error);
+    ApplicationError sqlError = <ApplicationError> dbError;
+    test:assertTrue(stringutils:contains(sqlError.message(), "Client is already closed"), sqlError.message());
+}
+
+function validateConnectionTimeoutError(int|error dbError) {
+    test:assertTrue(dbError is error);
+    DatabaseError sqlError = <DatabaseError> dbError;
+    test:assertTrue(stringutils:contains(sqlError.message(), "request timed out after"), sqlError.message());
+}
```
Comment (daneshk) [inline sql-ballerina/src/sql/tests/pool/connection-pool-test.bal:595]:
```suggestion
}

```

File: sql-ballerina/src/sql/tests/utils.bal [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+function queryMockClient(string url, @untainted string|ParameterizedQuery sqlQuery)
+returns @tainted record {}? {
+    MockClient dbClient = checkpanic new (url = url, user = user, password = password);
+    stream<record{}, error> streamData = dbClient->query(sqlQuery);
+    record {|record {} value;|}? data = checkpanic streamData.next();
+    checkpanic streamData.close();
+    record {}? value = data?.value;
+    checkpanic dbClient.close();
+    return value;
+}
```
Comment (daneshk) [inline sql-ballerina/src/sql/tests/utils.bal:99]:
```suggestion
}

```

**Review by daneshk — APPROVED** [review]
Add a few new lines suggestions, others LGTM

**Review by MadhukaHarith92 — APPROVED** [review]
Added a suggestion, rest LGTM.

File: sql-ballerina/build.gradle (hunk 1 above)
Comment (niveathika) [reply sql-ballerina/build.gradle:135]: I refactored the build file such that `./gradlew build -x test` will skip the tests.
https://github.com/ballerina-platform/module-ballerina-sql/blob/6192866ea89bb8936f8c7d2f04d7c5c21e441662/sql-ballerina/build.gradle#L94

**Pull request merged**